### PR TITLE
Redacting all sensitive information out of non stdout logs

### PR DESCRIPTION
**Pull request by juandiegopalomino**
# Description
Currently only removing the gcp project name or aws account id as opta doesn't show any other secrets.

# Safety checklist
* [x] This change is backwards compatible and safe to apply by existing users
* [x] This change will NOT lead to data loss
* [x] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
Ran opta apply manually with an injected error for both gcp and aws and, unlike before, now the crash report does not mention my account id/project at all.

### CODE REVIEW

**Review by NitinAgg — COMMENTED** [review]
Can you add some tests. Thanks

File: opta/utils/__init__.py [hunk 1]
```diff
@@ -21,6 +21,40 @@
...
+            r"([\"\: \'\n]?)project([\"\: \'\n]?): ([\"\: \'\n]?)[a-zA-Z0-9_\-]+([\"\: \'\n]?)",
+            "\g<1>project\g<2>: \g<3>REDACTED\g<4>",
+        ),
+        (r"projects/[a-zA-Z0-9_\-]+/", "projects/REDACTED/"),
+        (
+            r"@[a-zA-Z0-9_\-]+.iam.gserviceaccount.com",
+            "@REDACTED.iam.gserviceaccount.com",
+        ),
+        (r"[a-zA-Z0-9_\-]+.svc.id.goog", "REDACTED.svc.id.goog",),
+        (r"gcr.io/[a-zA-Z0-9_\-]+", "gcr.io/REDACTED"),
```
Comment (rabbitfang) [inline opta/utils/__init__.py:40]:
```suggestion
        (
            r"@[a-zA-Z0-9_\-]+\.iam\.gserviceaccount\.com",
            "@REDACTED.iam.gserviceaccount.com",
        ),
        (r"[a-zA-Z0-9_\-]+\.svc\.id\.goog", "REDACTED.svc.id.goog",),
        (r"gcr\.io/[a-zA-Z0-9_\-]+", "gcr.io/REDACTED"),
```

Treat the `.` in domain names as literals.

**Review by rabbitfang — CHANGES_REQUESTED** [review]
Tests should be added to this so we can verify that the redaction is happening as expected.

File: opta/crash_reporter.py [hunk 2]
```diff
@@ -79,5 +79,5 @@ def emit(self, record: LogRecord) -> None:
 CURRENT_CRASH_REPORTER = CrashReporter()
 crash_report_handler = ListHandler(CURRENT_CRASH_REPORTER.log_lines)
 crash_report_handler.setLevel(DEBUG)
-crash_report_handler.setFormatter(LogFormatMultiplexer())
+crash_report_handler.setFormatter(SensitiveFormatter())
```
Comment (legalosLOTR) [inline opta/crash_reporter.py:82]: Did you mean to replace LogFormatMultiplexer with SensitiveFormatter? Looks like you've already added SensitiveFormatter to the LogFormatMultiplexer class itself.

Comment (juandiegopalomino) [reply opta/crash_reporter.py:82]: Yes, this is intended

Comment (juandiegopalomino) [reply opta/crash_reporter.py:82]: We weren't multiplexing anyting there so it's fine

Comment (legalosLOTR) [reply opta/crash_reporter.py:82]: cool!

**Review by NitinAgg — APPROVED** [review]

**Review by rabbitfang — APPROVED** [review]

**Pull request merged**